streaming/streaming.go

```diff
@@ -157,25 +158,61 @@ func ReadStreamingConfig(file, dbName string) (StreamingCfg, error) {
 
 func getMysqlSourceStreamConfig(dbName string) *datastreampb.SourceConfig_MysqlSourceConfig {
 	mydb := &datastreampb.MysqlDatabase{
-		DatabaseName: dbName,
+		Database: dbName,
 	}
 	mysqlSrcCfg := &datastreampb.MysqlSourceConfig{
-		Allowlist: &datastreampb.MysqlRdbms{MysqlDatabases: []*datastreampb.MysqlDatabase{mydb}},
+		IncludeObjects: &datastreampb.MysqlRdbms{MysqlDatabases: []*datastreampb.MysqlDatabase{mydb}},
 	}
 	return &datastreampb.SourceConfig_MysqlSourceConfig{MysqlSourceConfig: mysqlSrcCfg}
 }
 
 func getOracleSourceStreamConfig(dbName string) *datastreampb.SourceConfig_OracleSourceConfig {
 	oracledb := &datastreampb.OracleSchema{
-		SchemaName: dbName,
+		Schema: dbName,
 	}
 	oracleSrcCfg := &datastreampb.OracleSourceConfig{
-		Allowlist: &datastreampb.OracleRdbms{OracleSchemas: []*datastreampb.OracleSchema{oracledb}},
+		IncludeObjects: &datastreampb.OracleRdbms{OracleSchemas: []*datastreampb.OracleSchema{oracledb}},
 	}
 	return &datastreampb.SourceConfig_OracleSourceConfig{OracleSourceConfig: oracleSrcCfg}
 }
 
-func getSourceStreamConfig(srcCfg *datastreampb.SourceConfig, sourceProfile profiles.SourceProfile) error {
+func getPostgreSQLSourceStreamConfig(properties string) (*datastreampb.SourceConfig_PostgresqlSourceConfig, error) {
+	params, err := profiles.ParseMap(properties)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse properties: %v", err)
+	}
+	infoschemadb := &datastreampb.PostgresqlSchema{
+		Schema: "infoschema",
+	}
+	pgcatalogdb := &datastreampb.PostgresqlSchema{
+		Schema: "pg_catalog",
+	}
+	pg_toast := &datastreampb.PostgresqlSchema{
+		Schema: "pg_toast",
+	}
+	postgres := &datastreampb.PostgresqlSchema{
+		Schema: "postgres",
+	}
+	pg_temp_1 := &datastreampb.PostgresqlSchema{
+		Schema: "pg_temp_1",
+	}
+	pg_toast_temp_1 := &datastreampb.PostgresqlSchema{
+		Schema: "pg_toast_temp_1",
```

Comment:
This bit can be refactored as a function that loops over the schema names and returns a []*datastreampb.PostgresqlSchema object. Then we simply call that instead of creating multiple objects.